static/panes/tool-input-view.ts

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2021, Tom Ritter
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import * as monaco from 'monaco-editor';
+import _ from 'underscore';
+import { MonacoPane } from './pane';
+import { ga } from '../analytics';
+import * as monacoConfig from '../monaco-config';
+import { Container } from 'golden-layout';
+import { MonacoPaneState } from './pane.interfaces';
+import { Hub } from '../hub';
+import { ToolInputViewState } from './tool-input-view.interfaces';
+import { Settings } from '../settings';
+
+export class ToolInputView extends MonacoPane<monaco.editor.IStandaloneCodeEditor, ToolInputViewState> {
+    source: string;
+    _toolId: number;
+    _toolName: string;
+    shouldSetSelectionInitially: boolean;
+    debouncedEmitChange: (() => void) & _.Cancelable;
+    cursorSelectionThrottledFunction: ((e: any) => void) & _.Cancelable;
+    lastChangeEmitted: string | undefined;
+    constructor(hub: Hub, container: Container, state: ToolInputViewState & MonacoPaneState) {
+        state = state || {}; // TODO according to TS typing this will be a no-op
+        if((state as any).compilerId) state.id = (state as any).compilerId;
+        super(hub, container, state);
+        this.source = state.source || '';
+    
+        this.settings = Settings.getStoredSettings();
+    
+        this._toolId = state.toolId;
+        this._toolName = state.toolName;
+        // TODO according to TS typing this should always be true
```

Comment:
This is probably a bug on the base pane. Selection is set to be `monaco.Selection` in `pane.interfaces.ts` so TS thinks it will never be true, so we should probably change that

Reply:
I marked this with a TODO because I wasn't sure right away what to do with it. `shouldSetSelectionInitially` seems somewhat unneeded, no other pane does this, but it was part of the js.
Other panes do check `if (this.selection)` so it would probably be good to examine the handling in the base pane.
It's a fun challenge that every .js pane does things slightly differently 😅

Reply:
Any thoughts on turning on `@typescript-eslint/no-unnecessary-condition` for the linter?

Reply:
By all means, do it! The more sensible linter settings that are on, the better